redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt

```diff
@@ -0,0 +1,2041 @@
+          getFlexItemMarginStartMain(child, isMainAxisHorizontal) +
+          getFlexItemMarginEndMain(child, isMainAxisHorizontal)
+        )
+      flexLine.totalFlexGrow += child.flexGrow
+      flexLine.totalFlexShrink += child.flexShrink
+      largestSizeInCross = max(
+        largestSizeInCross,
+        getViewMeasuredSizeCross(child, isMainAxisHorizontal) +
+          getFlexItemMarginStartCross(child, isMainAxisHorizontal) +
+          getFlexItemMarginEndCross(child, isMainAxisHorizontal),
+      )
+      // Temporarily set the cross axis length as the largest child in the flexLine
+      // Expand along the cross axis depending on the alignContent property if needed
+      // later
+      flexLine.crossSize = max(flexLine.crossSize, largestSizeInCross)
+      if (isMainAxisHorizontal) {
+        if (flexWrap != FlexWrap.WrapReverse) {
+          flexLine.maxBaseline = max(
+            flexLine.maxBaseline,
+            child.baseline + child.margin.top,
+          )
+        } else {
+          // if the flex wrap property is WRAP_REVERSE, calculate the
+          // baseline as the distance from the cross end and the baseline
+          // since the cross size calculation is based on the distance from the cross end
+          flexLine.maxBaseline = max(
+            flexLine.maxBaseline,
+            child.measuredHeight - child.baseline + child.margin.bottom,
+          )
+        }
+      }
+      if (isLastFlexItem(i, childCount, flexLine)) {
+        addFlexLine(flexLines, flexLine, i, sumCrossSize)
+        sumCrossSize += flexLine.crossSize
+      }
+      if (toIndex != -1 && flexLines.size > 0 &&
+        flexLines[flexLines.size - 1].lastIndex >= toIndex &&
+        i >= toIndex && !reachedToIndex
+      ) {
+        // Calculated to include a flex line which includes the flex item having the
+        // toIndex.
+        // Let the sumCrossSize start from the negative value of the last flex line's
+        // cross size because otherwise flex lines aren't calculated enough to fill the
+        // visible area.
+        sumCrossSize = -flexLine.crossSize
+        reachedToIndex = true
+      }
+      if (sumCrossSize > needsCalcAmount && reachedToIndex) {
+        // Stop the calculation if the sum of cross size calculated reached to the point
+        // beyond the needsCalcAmount value to avoid unneeded calculation in a
+        // RecyclerView.
+        // To be precise, the decoration length may be added to the sumCrossSize,
+        // but we omit adding the decoration length because even without the decorator
+        // length, it's guaranteed that calculation is done at least beyond the
+        // needsCalcAmount
+        break
+      }
+    }
+    return flexLines
+  }
+
+  /**
+   * Returns an array with container's padding in the format: [mainStart, mainEnd, crossStart, crossEnd].
+   */
+  private fun getPadding(isMainAxisHorizontal: Boolean): IntArray {
```

Comment:
This temp allocation is most likely going to be a performance tragedy ! (No action to take here; just whining)